src/regdesk/states/loading_state.js

```diff
@@ -0,0 +1,51 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * Loading state
+ */
+export class LoadingState extends RegState {
```

Comment:
I'm wondering if you're wanting to use this LoadingState in a general sense (like anytime there is a need to load) or only when specifically loading a result from, for example, search? If the former, I have concerns. If the latter, it should be named more specific :).

Edit: After reading the rest of the code it seems like this class is a mix, but slightly leaning more towards the latter.

Reply:
Correct, I expect this to end up as a dedicated state in between specifically performing a search for an identity and showing the results of those identities.

There's two other possible locations for a loading state: Hitting the API to mark someone paid, and hitting the API to check someone in. The paid status will likely end up being an aspect of the single result page anyway and not a dedicated state. Check-in will happen alongside badge print, and will likely also not be a separate state. Still hammering out details into the state diagram I'm building.

Reply:
Clarified class name.